### PR TITLE
Add doc comments to the `GitInformation` PD API

### PR DESCRIPTION
**Pull request by neonichu**
Should improve things like https://developer.apple.com/documentation/packagedescription/gitinformation/currentcommit?changes=latest_minor

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please test

File: Sources/PackageDescription/Context.swift [hunk 1]
```diff
@@ -47,7 +47,10 @@ public struct Context: Sendable {
 /// Information about the git status of a given package, if available.
 @available(_PackageDescription, introduced: 6.0)
 public struct GitInformation: Sendable {
+    /// The version tag currently checked out, if available.
     public let currentTag: String?
+    /// The commit currently checked out.
     public let currentCommit: String
+    /// Whether or not there are uncommited changes in the current repository.
```
Comment (bnbarham) [inline Sources/PackageDescription/Context.swift:54]:
```suggestion
    /// Whether or not there are uncommitted changes in the current repository.
```

**Review by bnbarham — APPROVED** [review]

**Comment by neonichu** [comment]
@swift-ci please test

**Comment by neonichu** [comment]
@swift-ci please test windows

**Review by MaxDesiatov — APPROVED** [review]

**Pull request merged**